src/shogun/machine/FeatureDispatchCRTP.h

```diff
@@ -0,0 +1,118 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Shubham Shukla
+ */
+
+#ifndef _FEATUREDISPATCHCRTP_H__
+#define _FEATUREDISPATCHCRTP_H__
+
+#include <shogun/lib/config.h>
+
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/lib/common.h>
+#include <shogun/machine/LinearMachine.h>
+
+namespace shogun
+{
+	class CFeatures;
+	class CLabels;
+#define IGNORE_IN_CLASSLIST
+
+	template <class P, class T>
+	IGNORE_IN_CLASSLIST class CDenseRealDispatch : public T
+	{
+	public:
+		/** Default constructor */
+		CDenseRealDispatch() : T()
+		{
+		}
+
+		virtual ~CDenseRealDispatch()
+		{
+		}
+
+	protected:
+		virtual bool train_dense(CFeatures* data)
+		{
+			auto obj = this->template as<P>();
+			switch (data->get_feature_type())
+			{
+			case F_DREAL:
+				return obj->template train_machine_templated<float64_t>(
+				    data->as<CDenseFeatures<float64_t>>());
+			case F_SHORTREAL:
+				return obj->template train_machine_templated<float32_t>(
+				    data->as<CDenseFeatures<float32_t>>());
+			case F_LONGREAL:
+				return obj->template train_machine_templated<floatmax_t>(
+				    data->as<CDenseFeatures<floatmax_t>>());
+			default:
+				SG_SERROR(
+				    "Training with DenseFeatures of provided %s type is not "
```

Comment:
Training with %s of provided type %s is not ....(move type and use get_name of features)